src/Common/src/System/Text/SimpleRegex.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Text
+{
+    internal static class SimpleRegex
+    {
+        // Based on wildcmp written by Jack Handy - <A href="mailto:jakkhandy@hotmail.com">jakkhandy@hotmail.com</A>
+        // https://www.codeproject.com/Articles/1088/Wildcard-string-compare-globbing
+
+        /// <summary>
+        /// Perform a match between an input string and a pattern in which the only special character
+        /// is an asterisk, which can map to zero or more of any character in the input.
+        /// </summary>
+        /// <param name="input">The input to match.</param>
+        /// <param name="pattern">The pattern to match against.</param>
+        /// <returns>true if the input matches the pattern; otherwise, false.</returns>
+        public static bool IsMatchWithStarWildcard(ReadOnlySpan<char> input, ReadOnlySpan<char> pattern)
+        {
+            int inputPos = 0, inputPosSaved = -1;
+            int patternPos = 0, patternPosSaved = -1;
+
+            // Loop through each character in the input.
+            while (inputPos < input.Length)
+            {
+                if (patternPos < pattern.Length && pattern[patternPos] == '*')
+                {
+                    // If we're currently positioned on a wildcard in the pattern,
+                    // move past it and remember where we are to backtrack to.
+                    inputPosSaved = inputPos;
+                    patternPosSaved = ++patternPos;
+                }
+                else if (patternPos < pattern.Length && char.ToLowerInvariant(pattern[patternPos]) == char.ToLowerInvariant(input[inputPos]))
```

Comment:
How cheap is `ToLowerInvariant` ? You could check without first.
`              else if (patternPos < pattern.Length && (pattern[patternPos] == input[inputPos] || char.ToLowerInvariant(pattern[patternPos]) == char.ToLowerInvariant(input[inputPos])))`

Reply:
While playing with my approach, I measured the above: it's 30% faster on your test set of inputs.

Reply:
else if (patternPos < pattern.Length && (
                    (pattern[patternPos] == input[inputPos] ||
                    char.ToLowerInvariant(pattern[patternPos]) == char.ToLowerInvariant(input[inputPos])))
                    )
```suggestion
                else if (patternPos < pattern.Length && (
                    (pattern[patternPos] == input[inputPos] ||
                    char.ToLowerInvariant(pattern[patternPos]) == char.ToLowerInvariant(input[inputPos])))
                    )
```

Reply:
> You could check without first

Seems reasonable.  It does mean that not matching inputs get a little slower, but not meaningfully.